services/painting-surface-defect-detection-model-service/app/routers/predict.py

```
@@ -0,0 +1,198 @@
+        # Base64 디코딩 (Pydantic에서 이미 검증됨)
+        image_data = base64.b64decode(data.image_base64)
+        
+        # 예측 수행
+        result = predict_anomaly(image_data, data.confidence_threshold)
+        
+        logger.info("Base64 prediction completed successfully")
+        return result
+        
+    except Exception as e:
+        logger.error(f"Error in base64 prediction: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Base64 prediction failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)